microprofile.ls/org.eclipse.lsp4mp.ls/src/main/java/org/eclipse/lsp4mp/services/MicroProfileValidator.java

```diff
@@ -161,6 +162,29 @@ private void validatePropertyValue(String propertyName, ItemMetadata metadata, P
 		}
 	}
 
+	/**
+	 * Adds a "Contains unexpected wildcard" diagnostic if:
+	 * 
+	 * 1. The property comes from an extension
+	 * 2. The property metadata contains a wildcard
+	 * 3. The corresponding property key in application.properties contains a wildcard 
+	 * 
+	 * In this case the corresponding property key should not have a wildcard
+	 * 
+	 * @param propertyKey the property key as it appears in application.properties
+	 * @param metadata    the corresponding property metadata
+	 * @param property    the property node
+	 */
+	private void validateWildCardProperty(String propertyKey, ItemMetadata metadata, Property property) {
+		if (!StringUtils.hasText(metadata.getExtensionName()) || !metadata.getName().contains("*")) {

```

Comment:
Here, I use `!StringUtils.hasText(metadata.getExtensionName()` to determine if the current property metadata is a user-defined property (like: `@ConfigProperty(name = "greeting.hello{*}")`) or not.

Checking `!StringUtils.hasText(metadata.getExtensionName()` is not the best way to do it, maybe there is another way? Maybe we can have some sort of flag in `ItemMetadata` that is set when the jdt ls extension scans properties?

Reply:
I have it as `contains("*")`  instead of `contains("{*}")` for now, because I think it would be interesting to cover this case:

```
kubernetes.annotations[*].key= test
```

Reply:
In my latest push, I am checking against `metadata.getSource()` instead.